compiler/rustc_ast_lowering/src/path.rs

```diff
@@ -277,7 +277,9 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             // See rustc_resolve::late::lifetimes::LifetimeContext::add_missing_lifetime_specifiers_label
             let elided_lifetime_span = if generic_args.span.is_empty() {
                 // If there are no brackets, use the identifier span.
-                path_span
+                // HACK: we use find_ancestor_inside to associate to properly suggest elided spans
+                //  in paths originating from macros.
```

Comment:
nit: grammar

also, what do macros have to do with this?

Reply:
so macros have nothing to #92324. But [`src/test/ui/in-band-lifetimes/elided-lifetimes.rs`](https://github.com/rust-lang/rust/blob/master/src/test/ui/in-band-lifetimes/elided-lifetimes.rs#L63) regresses if we don't return a subspan of the `path_span`.

Reply:
That's presumably why `segment.ident.span` was changed to `path_span` in #90446, but we still need a span that points to _just_ the segment, not the whole path. 

So this finds the segment's span inside of any macro if `path_span` inside of one.